change/@ni-nimble-components-c4841b4a-c42f-4d23-9748-857bbff400dd.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```

Comment:
Do we need any framework-specific integration or testing? If not it could still be good to do a one-time verification that it works by adding an icon header to the Blazor and Angular example apps.

Reply:
Great call! Turns out my implementation for setting the `slot` attribute in the .ctor for `TableColumnBase` was not proper, and instead needed to be moved to a `connectedCallback` override. The Angular app demonstrated the issue.

Reply:
As we discussed offline, let's add icon headers to the Angular and Blazor example apps. It's non-trivial to test for it but at least that will give us a chance of discovering regressions manually.